OpenRA.Game/Actor.cs

```diff
@@ -96,7 +87,14 @@ internal Actor(World world, string name, TypeDictionary initDict)
 
 				Info = world.Map.Rules.Actors[name];
 				foreach (var trait in Info.TraitsInConstructOrder())
+				{
 					AddTrait(trait.Create(init));
+
+					// Some traits rely on properties provided by IOccupySpace in their initialization,
+					// so we must ready it now, we cannot wait until all traits have finished construction.
+					if (trait is IOccupySpaceInfo)
+						OccupiesSpace = Trait<IOccupySpace>();
```

Comment:
Alright; I'll just revisit this when adding `ISingleOccurrenceTrait` and `IMultipleOccurrenceTrait`